Region/Region.cc

```diff
@@ -520,10 +520,22 @@ casacore::IPosition Region::XyOrigin() {
 const casacore::ArrayLattice<casacore::Bool>* Region::XyMask() {
     // returns boolean mask of xy region
     casacore::ArrayLattice<casacore::Bool>* mask;
+    bool extended_region_ok(false);
 
     if (_xy_region != nullptr) {
         // get extended region
-        auto extended_region = static_cast<casacore::LCExtension*>(_xy_region->toLCRegion(_coord_sys, _image_shape));
+        casacore::LCExtension* extended_region;
```

Comment:
no problem with this approach, I think it makes sense to handle the exceptions in here.